service/host.go

```diff
@@ -184,32 +174,24 @@ func (s *manager) shutdown(err error, reason string, exitCode *int) (bool, error
 	return true, err
 }
 
-// addModules adds the given modules to a service manager
-func (s *manager) addModules(modules ...ModuleCreateFunc) error {
-	for _, mcf := range modules {
-		mi := ModuleCreateInfo{
-			Host:  s,
-			Items: make(map[string]interface{}),
-		}
-
-		mods, err := mcf(mi)
-		if err != nil {
-			return err
-		}
-
-		if !s.supportsRole(mi.Roles...) {
-			ulog.Logger(_simpleCtx).Info(
-				"module will not be added due to selected roles",
-				"roles", mi.Roles,
-			)
-		}
-
-		for _, mod := range mods {
-			err = s.addModule(mod)
-		}
+func (s *manager) addModule(name string, module ModuleCreateFunc, options ...ModuleOption) error {
```

Comment:
Having worked through this, it doesn't make sense to have default names for modules - it completely messes with scoping. Part of the point of this change is to eventually be able to scope all relevant `service.Host` methods by module (if we fix https://github.com/uber-go/fx/issues/306), and you need a name to do so. The default names got to be completely problematic, and presented a chicken-or-egg problem with creating a ModuleInfo vs creating a Module.